### DISCUSSION TITLE
Form field not accepting input

### DISCUSSION THREAD

**Discussion by tsekeres** [nss-evening-cohort-14/evening-client-side#279] [Capstones!] [answered]
Tad Sekeres

Issue:
I am adding a second view that has CRUD in my app. my form will not accept new values in the fields. No typing or pasting. 

Side note: This Crud process and components are nearly identical to my first layer of CRUD. So I have no idea why the issue popped up with this one.

Errors:
No error, just lack of functionality

What have you googled?
1. https://stackoverflow.com/questions/34006333/cant-type-in-react-input-text-field
2. https://reactjs.org/docs/uncontrolled-components.html#default-values
3. https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/

What have you tried?
1. tried adding various versions of "this." to affect how it is doing the handleInputChange.
2. tried to understand the difference between Controlled and Uncontrolled inputs. 
3. tried to change value= to defaultValue=   *this allowed value to be entered in the field but changed how my key value pairs in firebase were getting sent and saved.

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with:
https://github.com/tsekeres/appy-trails-capstone/blob/setup/src/components/ResourceForm.js

**Comment by TrinityChristiana** [comment, marked as answer]
Because your [`handleInputChange`](https://github.com/tsekeres/appy-trails-capstone/blob/setup/src/components/ResourceForm.js#L28) function is looking at the value of an input's name attribute to know what key in [`resource`](https://github.com/tsekeres/appy-trails-capstone/blob/setup/src/components/ResourceForm.js#L18-L25) to update, the value of the name attribute on each input should match one of the keys identified in that state object. 

Each value on the input is looking at a key in the state object for updates. If the key the input is watching does not get updated, then the input's value will not get updated.

Example:
``` jsx
  // Here's your state obj
  const [resource, setResource] = useState({
    image: image || '',
    title: title || '',
    description: description || '',
    link: link || '',
    userId: admin.userId,
    firebaseKey: firebaseKey || null,
  });
...
  // Here's an input
  <Input
    name="resourceImage" // This name should match one of the keys in your object above
    id="resourceImage"
    value={resource.image}
    type="url"
    placeholder="Enter an Image URL"
    onChange={handleInputChange}
  />
```

**Reply by tsekeres** [reply]
Thanks Trinity. Spot on!

**Comment by drteresavasquez** [comment]
Had a ticket that addressed this from @cmeffley https://github.com/nss-evening-cohort-14/evening-client-side/discussions/245

**Reply by tsekeres** [reply]
I saw that. But his solutions did not address my issue.

**Comment by tsekeres** [comment]
Soooo. Apparently its important what you typey type into EVERYTHING! As you may have read in Trinity's solution, I did not keep consistent naming for all of the input values. So the handleInputChange was not looking for the proper value and therefore not changing when my typing was happening.
